modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/SlimFrameworkServerCodegen.java

```diff
@@ -0,0 +1,195 @@
+        );
+
+        additionalProperties.put(CodegenConstants.INVOKER_PACKAGE, invokerPackage);
+        additionalProperties.put(CodegenConstants.GROUP_ID, groupId);
+        additionalProperties.put(CodegenConstants.ARTIFACT_ID, artifactId);
+        additionalProperties.put(CodegenConstants.ARTIFACT_VERSION, artifactVersion);
+
+        // ref: http://php.net/manual/en/language.types.intro.php
+        languageSpecificPrimitives = new HashSet<String>(
+                Arrays.asList(
+                        "boolean",
+                        "int",
+                        "integer",
+                        "double",
+                        "float",
+                        "string",
+                        "object",
+                        "DateTime",
+                        "mixed",
+                        "number")
+        );
+
+        instantiationTypes.put("array", "array");
+        instantiationTypes.put("map", "map");
+
+        // ref: https://github.com/swagger-api/swagger-spec/blob/master/versions/2.0.md#data-types
+        typeMapping = new HashMap<String, String>();
+        typeMapping.put("integer", "int");
+        typeMapping.put("long", "int");
+        typeMapping.put("float", "float");
+        typeMapping.put("double", "double");
+        typeMapping.put("string", "string");
+        typeMapping.put("byte", "int");
+        typeMapping.put("boolean", "boolean");
+        typeMapping.put("date", "DateTime");
+        typeMapping.put("datetime", "DateTime");
+        typeMapping.put("file", "string");
+        typeMapping.put("map", "map");
+        typeMapping.put("array", "array");
+        typeMapping.put("list", "array");
+        typeMapping.put("object", "object");
+
+        supportingFiles.add(new SupportingFile("README.mustache", packagePath.replace('/', File.separatorChar), "README.md"));
+        supportingFiles.add(new SupportingFile("composer.json", packagePath.replace('/', File.separatorChar), "composer.json"));
+        supportingFiles.add(new SupportingFile("index.mustache", packagePath.replace('/', File.separatorChar), "index.php"));
+        supportingFiles.add(new SupportingFile(".htaccess", packagePath.replace('/', File.separatorChar), ".htaccess"));
+    }
+
+    public CodegenType getTag() {
+        return CodegenType.SERVER;
+    }
+
+    public String getName() {
+        return "slim";
+    }
+
+    public String getHelp() {
+        return "Generates a Slim Framework server library.";
+    }
+
+    @Override
+    public String escapeReservedWord(String name) {
+        return "_" + name;
+    }
+
+    @Override
+    public String apiFileFolder() {
+        return (outputFolder + "/" + apiPackage()).replace('/', File.separatorChar);
+    }
+
+    public String modelFileFolder() {
+        return (outputFolder + "/" + modelPackage()).replace('/', File.separatorChar);
+    }
+
+    @Override
+    public String getTypeDeclaration(Property p) {
```

Comment:
I would recommend reusing the one in https://github.com/swagger-api/swagger-codegen/blob/master/modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/PhpClientCodegen.java#L228